packages/components/src/Dialog/Layout/DialogLayout.story.tsx

```diff
@@ -74,16 +68,51 @@ Full.args = {
   ...Header.args,
 }
 
-const NoPaddingTemplate: Story<DialogLayoutExampleProps> = () => (
+const extraButton = (
+  <ButtonTransparent color="neutral" size="medium">
+    Cancel
+  </ButtonTransparent>
+)
+
+export const HeaderDetail = () => (
```

Comment:
This story could leverage the standard `Template` above by being rewritten as:

```suggestion
export const HeaderDetail = Template.bind({})
HeaderDetail.args = {
  ...Content.args,
  header: 'Header',
  headerDetail: 'Header detail text',
}
```

If you need to leverage the complete story in a Jest test you can do so by following this article: https://storybook.js.org/docs/react/workflows/unit-testing

TLDR: 

```
it('renders the button in the primary state', () => {
  render(<HeaderDetail {...HeaderDetail.args} />);
  expect(screen.getByText('Header detail text');
});
```

Note the `{...HeaderDetail.args} ` that spreads the args onto the component is a key piece here.

Reply:
Why should we do this change?